pyat/at/collective/beam_loading.py

```diff
@@ -135,16 +146,21 @@ def __init__(self, family_name: str, length: float, voltage: float,
             Nturns (int):       Number of turn for the wake field. Default: 1
             ZCuts:              Limits for fixed slicing, default is adaptive
             NormFact (float):   Normalization factor
-            mode (BLMode):  method for beam loading calculation BLMode.PHASOR
+            blmode (BLMode):  method for beam loading calculation BLMode.PHASOR
                 (default) uses the phasor method, BLMode.WAKE uses the wake
                 function. For high Q resonator, the phasor method should be
                 used
+            cavitymode (CavityMode):  Is cavity ACTIVE (default) or PASSIVE
         Returns:
             bl_elem (Element): beam loading element
         """
         kwargs.setdefault('PassMethod', self.default_pass[True])
-        assert isinstance(mode, BLMode), \
-            'Beam loading mode has to be an instance of BLMode'
+        if not isinstance(blmode, BLMode):
+            raise TypeError('Beam loading mode has to be an' +
+                            'instance of BLMode')
+        if not isinstance(cavitymode, CavityMode):
+            raise TypeError('Cavity mode has to be an' +
```

Comment:
raise AtError() and use cavitymode

Reply:
Can you give an  example usage? AtError is fine but use cavitymode is the message?

Reply:
Ah sorry, for this one and all the others it is better to put the argument name in the error message.
I would then put:
`raise TypeError('cavitymode has to be an instance of CavityMode')`

Reply:
got it